### DISCUSSION TITLE
Keysight App

### DISCUSSION THREAD

**Discussion by NoiZeBeAsT** [onlaj/Piano-LED-Visualizer#234] [Ideas]
I recently discovered the (desktop) app Keysight. It's really good and works well together with the LEDs.
But it has some features I can't implement with the visualizer (or I haven't figured out how to):

* Sequences triggered not by pedals but the key itself. I mean I hit any key and it jumps to the next sequence. That would allow me to have alternating colors.

* Multiple color stops on the gradients so you could go red - blue - red, for example.

The Keysight app supports websockets to send/reveice data like the color of a key etc. maybe there is a way to have the RPi connect to the websocket and use the data on the fly?

**Comment by onlaj** [comment]
>Sequences triggered not by pedals but the key itself. I mean I hit any key and it jumps to the next sequence. That would allow me to have alternating colors.

I think It would be very chaotic to change colors with every piano key press, in what scenario you think it would be useful?

>Multiple color stops on the gradients so you could go red - blue - red, for example.

You can use multicolor for something like that, I know it's not ideal but I really don't want to overcomplicate color settings.

>The Keysight app supports websockets to send/reveice data like the color of a key etc. maybe there is a way to have the RPi connect to the websocket and use the data on the fly?

I took a look at it and it seems that the author doesn't care that much about websockets feature on his app. https://steamcommunity.com/sharedfiles/filedetails/?id=2218116335 I don't want to spend my time on feature that works only with one third-party app and can break when that app is updated.

**Reply by NoiZeBeAsT** [reply]
Sorry for my late reply.

> I think It would be very chaotic to change colors with every piano key press, in what scenario you think it would be useful?

I think I've seen this in some of the youtube videos and I'm pretty sure Keysight app has a feature like that as well.
Chaotic could make some nice effects while playing staccato (Joja Wendt's Asturias for example).

> You can use multicolor for something like that, I know it's not ideal but I really don't want to overcomplicate color settings.

That would require a preset with 88 different color settings right? I might be able to write a script to automate that.

> I took a look at it and it seems that the author doesn't care that much about websockets feature on his app. https://steamcommunity.com/sharedfiles/filedetails/?id=2218116335 I don't want to spend my time on feature that works only with one third-party app and can break when that app is updated.

To me it sounds more like he is lacking the knowledge to do it. What a pity, your LEDs and Keysight seem to be made for each other. :) But I totally understand your point.
